src/src/components/Contexts/SettingsContext.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { ajax } from '../../utils/ajax';
+import { AuthContext } from './AuthContext';
+
+type UserSettings = {
+  readonly weekStart: number;
+};
+
+export function SettingsProvider({
+  children,
+}: {
+  readonly children: React.ReactNode;
+}): JSX.Element {
+  const authContext = React.useContext(AuthContext);
+  const [weekStart, setWeekStart] = React.useState<number>(0);
+  const [found, setFound] = React.useState<boolean>(false);
+
+  // If the token has been set, try an update
+  // Also make sure we havent already fetched
+  if (authContext.token && !found) {
+    ajax(
+      'https://www.googleapis.com/calendar/v3/users/me/settings/weekStart'
+    ).then((res) => {
+      res.json().then((val) => {
```

Comment:
Can refactor to be:
```
.then((response)=>response.json())
.then((response)=>{

})
```

(i.e, flatten it, rather than build a pyramid out of `then` calls